### PR TITLE
feat: Add audit logs filtering to the UI

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Related to #4077

Preview: 
image

### CODE REVIEW

File: docs/admin/audit-logs.md [hunk 1]
```diff
@@ -13,6 +13,18 @@ This feature tracks **create, update and delete** events for the following resou
...
+In the Coder UI you can filter your audit logs using the pre-defined filter or by using the Coder's filter query like the examples below:
+
+- `resource_type:workspace action:delete` to find deleted workspaces
+- `resource_type:template action:create` to find created templates
+
+The supported filters are:
+
+- `resource_type` - The type of the resource. It can be a workspace, template, user, etc. You can [find here](https://pkg.go.dev/github.com/coder/coder@main/codersdk#ResourceType) all the resource types that are supported.
+- `action`- The action applied to a resource. You can [find here](https://pkg.go.dev/github.com/coder/coder@main/codersdk#AuditAction) all the actions that are supported.
+
```
Comment (BrunoQuaresma) [inline docs/admin/audit-logs.md:27]: @bpmct I would appreciate your help with this doc

File: site/src/pages/AuditPage/AuditPage.test.tsx [hunk 2]
```diff
@@ -33,4 +26,40 @@ describe("AuditPage", () => {
...
+        .spyOn(API, "getAuditLogs")
+        .mockResolvedValue({ audit_logs: [MockAuditLog] })
+
+      const query = "resource_type:workspace action:create"
+      history.push(`/audit?filter=${encodeURIComponent(query)}`)
+      render(<AuditPage />)
+
+      await waitForLoaderToBeRemoved()
+
+      expect(getAuditLogsSpy).toBeCalledWith({ limit: 25, offset: 0, q: query })
```
Comment (jsjoeio) [inline site/src/pages/AuditPage/AuditPage.test.tsx:62]: nice tests 👏🏼

**Review by jsjoeio — APPROVED** [review]
Nice improvement!

File: docs/admin/audit-logs.md (hunk 1 above)
Comment (bpmct) [reply docs/admin/audit-logs.md:27]: This works well for now! Sorry I missed it

**Pull request merged**